app/src/main/java/de/westnordost/streetcomplete/screens/about/ChangelogFragment.kt

```diff
@@ -52,20 +51,21 @@ class WhatsNewDialog(context: Context, sinceVersion: String) : AlertDialog(conte
 
     init {
         val binding = DialogWhatsNewBinding.inflate(LayoutInflater.from(context))
-        binding.changelogList.addItemDecoration(DividerItemDecoration(context, DividerItemDecoration.VERTICAL))
 
         setTitle(R.string.title_whats_new)
         setView(binding.root)
         setButton(DialogInterface.BUTTON_POSITIVE, context.resources.getText(android.R.string.ok), null, null)
 
         scope.launch {
             val fullChangelog = readChangelog(context.resources)
-            var currentVersionIndex = fullChangelog.indexOfFirst { it.title == sinceVersion }
-            // if version not found, just show the last one
-            if (currentVersionIndex == -1) currentVersionIndex = 1
-            val changelog = fullChangelog.subList(0, currentVersionIndex)
-
-            binding.changelogList.adapter = ChangelogAdapter(changelog)
+            var sinceVersionIndex = fullChangelog.indexOf("<h2>$sinceVersion</h2>")
+            if (sinceVersionIndex == -1) {
+                // if version not found, just show the last one
+                sinceVersionIndex = fullChangelog.indicesOf("<h2>").elementAt(1)
```

Comment:
throws exception if there is no element at 1?

Reply:
Now it does, see [`99aba20`](https://github.com/streetcomplete/StreetComplete/pull/5439/commits/99aba20c10349a4f1e16bc5cf99aa2a7eeec32bf).

Reply:
Actually, my point was that it should not throw an exception in case there are not two `<h2>` headlines in the file. But well, not so important because there are more than two headlines in the file.

Reply:
And, I guess that was the behavior before, too, so my fault